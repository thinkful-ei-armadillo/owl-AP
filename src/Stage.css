.stage-container {
    display: flex;
    flex-wrap: wrap;
    text-align: center;
}

.info-status {
    color: white;
}