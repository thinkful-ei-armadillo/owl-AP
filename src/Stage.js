import React from 'react';
import StageParticipant from './StageParticipant';
import './Stage.css'


export default function Stage(props) {
    const stageParticipants = props.participants
        .filter(element => element.onStage)
        .map(element => 
            <StageParticipant 
            key={element.id} 
            name={element.name} 
            avatar={element.avatar}
            self={element.self || false} 
            />
            );

    return (
        <div className="stage-container">
            {stageParticipants}
        </div>
    );
}