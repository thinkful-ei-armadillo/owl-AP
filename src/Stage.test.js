import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';
import store from './store';

describe('ParticipantList component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Stage participants={store.participants} />, div);

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Stage participants={store.participants} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
