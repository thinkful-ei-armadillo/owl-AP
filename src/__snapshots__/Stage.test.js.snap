// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList component testing renders the UI as expected 1`] = `
<div
  className="stage-container"
>
  <div
    className="stage-participant-self"
  >
    <div
      className="info-status"
    >
      Koren Templeton
    </div>
    <img
      alt="Koren Templeton"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="info-status"
    >
      Caty Flucker
    </div>
    <img
      alt="Caty Flucker"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="info-status"
    >
      Ashla Attwool
    </div>
    <img
      alt="Ashla Attwool"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
